### ISSUE TITLE
project.pbxproj: omit quotes around MARKETING_VERSION and CURRENT_PROJECT_VERSION

### ISSUE THREAD

**Issue by rofe** [alexandermendes/semantic-release-react-native#4]
A while ago, XCode started to write `MARKETING_VERSION` and `CURRENT_PROJECT_VERSION` without quotes in `project.pbxproj`. This leads to merge conflicts in GitHub in branches where changes were made in XCode , e.g.

```
<<<<<<< branch-xyz
        MARKETING_VERSION = 6.23.0;
=======
	MARKETING_VERSION = "6.25.1";
>>>>>>> main
```

Is there a way to omit the quotes? My current config looks like this:
```js
{
  plugins: [
    ...
    ['semantic-release-react-native', {
      'versionStrategy': {
        'ios': { 'buildNumber': 'semantic' },
      },
      'iosPath': 'src/foo',
      'skipAndroid': true,
    }],
    ['@semantic-release/git', {
        'src/foo/foo.xcodeproj/project.pbxproj'
      ],
      'message': 'chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}',
    }],
  ],
}
```

**Comment by alexandermendes** [comment]
Interesting, probably possible but not totally straightforward to fix that, as that version number is written from a JavaScript object (using [pbxproj-dom](https://www.npmjs.com/package/pbxproj-dom)):

https://github.com/alexandermendes/semantic-release-react-native/blob/369f65e429180e7bd68d00bdb020818f9a11b887/src/version/ios.ts#L167C1-L177

and the JavaScript isn't going to like it if we try to define `1.2.3` not as a string. Ideas welcome though!

**Comment by rofe** [comment]
@alexandermendes I see. This is certainly not the most elegant solution, but gets the job done:
```js
  xcode.save();
  // strip double quotes from version strings
  const versionRE = new RegExp(`([${currentProjectVersionKey}|${marketingVersionKey}]) = "(.*)"`, 'gm');
  const xcodeDoc = fs
    .readFileSync(xcode.path, { encoding: 'utf-8' })
    .replaceAll(versionRE, '$1 = $2');
  fs.writeFileSync(xcode.path, xcodeDoc, { encoding: 'utf8' });

```

**Comment by alexandermendes** [comment]
Seems reasonable, thanks! Should be fixed with [v1.8.1](https://github.com/alexandermendes/semantic-release-react-native/releases/tag/v1.8.1).

**Comment by alexandermendes** [comment]
Will address the remaining issue in https://github.com/alexandermendes/semantic-release-react-native/issues/5
